module-4/backend/week_two.md

```diff
@@ -1,22 +1,34 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's one difference between ES5 and ES6?
-2. What's the difference between asynchronous and synchronous JavaScript? 
-3. What are the four pillars of Object Oriented programming?
-4. What are some tools available in JavaScript to help you write object oriented code?
-5. What are some key concepts to be aware of when refactoring your JavaScript?
-6. What's a callback function and what are some reasons when we use/need callback functions?
-7. What's the scope of variables in Javascript?
-8. What's the difference between `==` and `===` in JavaScript?
-9. Why do front end frameworks exist?
+ES6 supports classes. Also fat arrow syntax.
+
+2. What's the difference between asynchronous and synchronous JavaScript?  Async: it doesn't flood the stack. When an async task is called within a function, if the stack isn't empty, it goes to the Web API, which then sends it to the event queue (after how long? does the web api immediately send the task to the event queue?). Once the stack is empty, the event loop will push the task from the queue to the stack. The benefit is that later tasks don't get backlogged behind that first task.
```

Comment:
BOOM! As far 'after how long?' - as soon as it resolves, so yes, immediately upon completely it's job!

I would just make sure to be clear that JS is a synchronous language by nature, and we have some functions that behave in an asynchronous manner.